flexmeasures/data/schemas/generic_assets.py

```diff
@@ -109,23 +109,41 @@ def validate_account(self, account_id: int | None):
             )
 
     @validates("attributes")
-    def validate_attributes(self, attributes: dict):
+    def validate_attributes(self, attributes: dict):  # noqa C901
         sensors_to_show = attributes.get("sensors_to_show", [])
 
         # Check type
         if not isinstance(sensors_to_show, list):
             raise ValidationError("sensors_to_show should be a list.")
         for sensor_listing in sensors_to_show:
-            if not isinstance(sensor_listing, (int, list)):
+            if not isinstance(sensor_listing, (int, list, dict)):
                 raise ValidationError(
-                    "sensors_to_show should only contain sensor IDs (integers) or lists thereof."
+                    "sensors_to_show should only contain sensor IDs (integers), lists of sensor IDs, or dictionaries with 'title' and 'sensor' keys."
                 )
             if isinstance(sensor_listing, list):
                 for sensor_id in sensor_listing:
                     if not isinstance(sensor_id, int):
                         raise ValidationError(
-                            "sensors_to_show should only contain sensor IDs (integers) or lists thereof."
+                            "sensors_to_show should only contain sensor IDs (integers), lists of sensor IDs, or dictionaries with 'title' and 'sensor' keys."
                         )
+            if isinstance(sensor_listing, dict):
```

Comment:
When formulating this as a `SensorsToShowSchema`, one inspiration could be the schema `TimeSeriesOrSensor` (see data/schemas/sensors.py), which also describes something that can come in many forms. It does this in the main `_deserialize()` function, which we should use to turn incoming data into objects.

Reply:
i think, i addressed that in new commits.